pkg/acceptance/bettertestspoc/assert/objectassert/task_snowflake_ext.go

```diff
@@ -0,0 +1,74 @@
+package objectassert
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"slices"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+)
+
+func (t *TaskAssert) HasNotEmptyCreatedOn() *TaskAssert {
+	t.AddAssertion(func(t *testing.T, o *sdk.Task) error {
+		t.Helper()
+		if o.CreatedOn == "" {
+			return fmt.Errorf("expected created on not empty; got: %v", o.CreatedOn)
+		}
+		return nil
+	})
+	return t
+}
+
+func (t *TaskAssert) HasNotEmptyId() *TaskAssert {
+	t.AddAssertion(func(t *testing.T, o *sdk.Task) error {
+		t.Helper()
+		if o.Id == "" {
+			return fmt.Errorf("expected id not empty; got: %v", o.CreatedOn)
+		}
+		return nil
+	})
+	return t
+}
+
+func (t *TaskAssert) HasPredecessors(ids ...sdk.SchemaObjectIdentifier) *TaskAssert {
```

Comment:
nit pick: in such assertions (when we require these two lists to match exactly without taking order into account - but this does not matter) it is useful to not only list the missing ids but also list the unexpected ones (and this is how it is usually implemented in the assertion libraries)

Reply:
Changed the assertions, and added InAnyOrder suffix to this function (on failure all items are printed, so It will be known what was an unexpected item).